### DISCUSSION TITLE
Desaturate image

### DISCUSSION THREAD

**Discussion by TheSpartaPT** [ImageMagick/ImageMagick#4921] [Help] [answered]
I'm trying to replicate the result given by photoshop when you desaturate an image (SHIFT + CTRL + U),
I've tried a bunch of ways already and they all convert the output image to a grayscale image and the colors on the output aren't exactly the same either,
is there any way to do this that keeps it as an RGB image ?

---

some of the ways I've tried already:

`magick input.png -colorspace Gray output.png` - converts to Grayscale, colors aren't the same
`magick input.png -type Grayscale output.png` - converts to Grayscale, colors aren't the same
`magick input.png -modulate 100,0 output.png` - converts to Grayscale, colors aren't the same

**Comment by fmw42** [comment, marked as answer]
This seems to be close for me in IM 6.9.12.42 on Mac OSX Sierra

Input:

![mandril3](https://user-images.githubusercontent.com/13458218/157311608-d1c57ff3-c698-47c9-bbc7-52ce0a44c1ab.jpg)


Photoshop Desat Saved to PNG:

![mandril3_ps_desat](https://user-images.githubusercontent.com/13458218/157309731-0c97f8ee-d772-4096-9b0c-47d8d5861e85.png)

IM Desat:

```
convert mandril3.jpg -modulate 100,0,100 x1.png
```

IM result :

![x1](https://user-images.githubusercontent.com/13458218/157309841-f4641447-3776-42cd-9613-060877523c1b.png)

Comparison:

```
compare -metric rmse mandril3_ps_desat.png x1.png null:
504.673 (0.00770082)

That is 0.8% different. Not bad!

```

**Reply by TheSpartaPT** [reply]
i tried the same compare command, and it is giving me a 0,01% difference between what I exported from photoshop and what I got from imagemagick, the problem is, i can tell there's a huge difference just by looking at it, the colors on the image exported from photoshop are wayyyyy darker than what i got from imagemagick

i'm using ImageMagick 7.1.0-26 btw

**Reply by fmw42** [reply]
0.01% is even better than what I got. I think your issue is likely viewing profiles assigned.  Be sure you use the same profile when saving files.  Photoshop may be using Adobe RGB and Imagemagick is likely using sRGB profiles.  Your viewer may also be assigning a different profile, especially if you do not assign a profile.  (What output format are you using?  Can you post your results?)

Also check the verbose information about the images to see what is different, esp the R,G,B statistics. Those should tell you that the images are similar.  You can also use EXIFTOOL to view the image meta data to see what might be different.

Also after saturating, there are no colors. So I do not understand what you mean. Are you actually saying the gray shades are different?  As in my result above, the two output images look the same to me. I do not see a huge difference.

**Reply by TheSpartaPT** [reply]
here's the desaturated images I got both from photoshop and imagemagick
https://thesparta.pt/images/desaturated_photoshop.png
https://thesparta.pt/images/desaturated_imagemagick.png

they are both 1024x1024, that's why I didn't just paste them here in the middle of the text 😅

viewing them through the browser the difference is subtle, I was comparing them in photoshop when I said there was a substantial difference

https://thesparta.pt/images/photoshop_screenshot.png

you can clearly see the left one looks darker than the right one

**Reply by fmw42** [reply]
Please post the original before desaturating

**Reply by fmw42** [reply]
When I download and save your two images, and view in Mac Preview, they look the same. I suspect that Photoshop is providing different profiles to each one.

**Reply by TheSpartaPT** [reply]
here's the original: https://thesparta.pt/images/original.png

**Reply by fmw42** [reply]
When I save your images to my desktop and then load them to Photoshop, Imagemagick one gets a gray default dot gain profile and the photoshop one gets a color sRGB (or Adobe RGB) profile.  So there are differences in profiles if you just take the defaults. Please check what profiles are being use by PS when viewing these two results.

**Reply by fmw42** [reply]
To get PS to assign a color profile with the IM result, try

`convert original.png -modulate 100,0,100 PNG24:original_im.png`

Then when you load your original in PS, use the sRGB profile and load the IM result with the sRGB profile.  Now do they look the same?

**Reply by TheSpartaPT** [reply]
yup that did it, thank you so much
